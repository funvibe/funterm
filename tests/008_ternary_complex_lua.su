lua.a = 5
lua.b = 5
lua.c = 10

# Простое равенство
lua.result1 = lua.a == lua.b ? "equal" : "not equal"
lua.print("Simple equality:", lua.result1)

# Сложное условие с &&
lua.result2 = (lua.a == lua.b) && (lua.c > 8) ? "both true" : "not both true"
lua.print("Complex AND:", lua.result2)

# Сложное условие с ||
lua.result3 = (lua.a > 10) || (lua.b == 5) ? "at least one true" : "both false"
lua.print("Complex OR:", lua.result3)

# Вложенные условия
lua.result4 = (lua.a == lua.b) && ((lua.c > 5) && (lua.c < 15)) ? "nested true" : "nested false"
lua.print("Nested conditions:", lua.result4)
