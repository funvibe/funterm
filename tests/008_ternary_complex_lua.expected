Simple equality: equal
Complex AND: both true
Complex OR: at least one true
Nested conditions: nested true
